src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/ConsoleLifetimeExitTests.cs

```diff
@@ -18,14 +21,22 @@ public class ConsoleLifetimeExitTests
         /// and the rest of "main" gets executed.
         /// </summary>
         [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
-        [PlatformSpecific(TestPlatforms.AnyUnix)]
         [InlineData(SIGTERM)]
         [InlineData(SIGINT)]
         [InlineData(SIGQUIT)]
```

Comment:
For SIGINT and SIGQUIT, you can pass `CreateNewProcessGroup` to RemoteInvokeOptions.StartInfo.  Summarizing, it will require that you 
* ifdef for just net10
* Add the necessary pinvokes 
* for SIGINT (CTRL_C_EVENT), you would need to call [SetConsoleCtrlHandler(null, false)](https://github.com/dotnet/runtime/blob/1df4e5231ea9ce2efac8dbc0eaede7cd4a928324/src/libraries/System.Diagnostics.Process/tests/ProcessTests.Windows.cs#L48-L57) in the remote process.
* Exclude the test when running Windows as a container.

It will be quite convoluted, but it will use native Windows console events.

Reply:
Yeah, I understand that.  We discussed doing that back in June before the CreateNewProcessGroup API was ready.  IMO that's functional testing, but this aims to unit test.  We also can't test shutdown in that way, unfortunately, which is the case this PR aims to fix.